actions/agent-connector/agent_api_client.py

```diff
@@ -18,38 +39,50 @@ def _get_api_url(self) -> str:
         Raises:
             ConnectionError: If no API server is responding
         """
-
         def test_url(url: str) -> bool:
             try:
                 parsed = urlparse(url)
                 if parsed.scheme not in ("http", "https"):
                     return False
-                sema4ai_http.get(f"{url}", timeout=1)
+                response = sema4ai_http.get(f"{url}", timeout=1)
                 return True
-            except (HTTPError, ValueError):
+            except (HTTPError, ValueError) as e:
+                return False
+            except Exception as e:
                 return False
 
         # First check environment variable
-        if url := os.getenv("SEMA4AI_AGENT_SERVER_API_URL"):
-            if test_url(url):
-                return url
-
-        # Try default ports
-        for port in [8990, 8000]:
-            url = f"http://localhost:{port}/api/v1"
-            if test_url(url):
-                return url
+        if url := os.getenv("SEMA4AI_TENANT_URL"):
+            # Append /api/v1 to the base URL for cloud environments
+            api_url = f"{url}/api/v1"
+            if test_url(api_url):
+                return api_url
+
+        # Try reading from agent-server.pid file
+        pid_file_path = os.path.expanduser("~/.sema4ai/sema4ai-studio/agent-server.pid")
```

Comment:
I think can fail on Windows. Maybe something like
`pid_file_path = os.path.join(os.path.expanduser("~"), ".sema4ai", "sema4ai-studio", "agent-server.pid")`

Reply:
Is the file even in the same location?

Reply:
Fixed on windows. Thanks @mikahanninen!